build/elasticsearch/bin/docker-entrypoint.sh

```diff
@@ -1,6 +1,40 @@
 #!/bin/bash
+set -e
 
-# Run Elasticsearch and allow setting default settings via env vars
+# Files created by Elasticsearch should always be group writable too
+umask 0002
+
+# Create random user entry, if different from uid 1000, to support Openshift
+# https://docs.openshift.org/latest/creating_images/guidelines.html
+if ! whoami &> /dev/null; then
+    if [ -w /etc/passwd ]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:/usr/share/elasticsearch:/sbin/nologin" >> /etc/passwd
```

Comment:
I don't think this user should be called `default`. The default user is `elasticsearch`. This user is the opposite of default. It's an unusual, unexpected and custom user. Let's call it `custom_user` or something.

From a different angle, did you consider changing the UID of the existing `elasticsearch` user instead? Is that a bad idea?

Reply:
> From a different angle, did you consider changing the UID of the existing elasticsearch user instead? Is that a bad idea?

That's a good suggestion. One of my concerns was for people who want to inspect a running container and a new user (with something like `--user <random> -ti <container_id> /bin/bash`) .

But this is very much a corner case I think and testing it seems to work fine. Therefore I suggest an even bolder move: remove this entire section. I introduced it as suggested in the [Openshift docs](https://docs.openshift.org/latest/creating_images/guidelines.html) for fear of applications that want to lookup the associated user from `/etc/passwd`. I tried building things again now without it and every combination I tried doesn't give problems, including entering the container with an interactive shell. I also tried some of the binaries under `bin/` and `bin/x-pack` and didn't see issues.

Adding one test case that attempts to run elasticsearch with `--user=<random` will cover us somewhat, but naturally we can't write test cases to execute every binary in this scenario.

I vote for removing it altogether, WDYT?

Reply:
👍 to that. Less code is less complexity, and I'm always in favour of that.

Also, I think having an anonymous user that's identified only by UID makes it very clear that is a local, ephemeral user.